bin/add_celltypes_to_sce.R

```diff
@@ -508,39 +508,60 @@ if (assign_consensus) {
         unique()
     }
 
-    stopifnot(
-      "A single broad diagnosis for the library could not be identified" = length(broad_diagnosis) == 1,
-      "Could not determine a single known diagnosis group from sample diagnosis." =
-        broad_diagnosis %in% diagnosis_celltype_df$diagnosis_group
-    )
+    # if the only broad diagnosis is non-cancerous then skip counting
+    # otherwise, proceed with counting
+    if (broad_diagnosis == "non-cancerous") {
+      # no reference cells here; store as character for writing to file
+      reference_cell_count <- "0"
```

Comment:
this should be fine as the number itself
```suggestion
      reference_cell_count <- 0
```